Iss11 medical abstracts

alamine42 opened the pull request:
There is a need to be able to pull medical abstracts from an online DB like PubMed in order to build or enrich the medical corpus.

The script med_abstract_ingest.py behaves exactly the same way as wikipedia_ingest.py. It takes three parameters:
- Search term: some topic keyword for the online search
- Data directory: which directory does the corpus reside in (if it doesn't exist, it will get created, otherwise files will just be added to it)
- Results: how many result hits should the script return. Each will be saved in its own .txt file in the data directory

Usage example:

```
python med_abstract_ingest.py -s cancer -d cancer -r 5
```

The above will save 5 medical abstracts related to cancer in a directory named 'cancer' under data/

There is also a configuration file: ingestion_config.py
It is meant to allow for configuration of any parameters related to the ingestion process (not just exclusive to med_abstract_ingest.py). Right now, it contains one section:

```
[medical_abstracts]
pubmed_search_url=http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=pubmed&term=<SEARCH_TERM>&retmax=<RESULTS>
pubmed_doc_url=http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=pubmed&id=<DOC_ID>&rettype=abstract&retmode=text
```

The above 2 URLs are used in the search. The first is used to get a list of document IDs related to the search term. The second is used to fetch each individual document text. Note the placeholder params <SEARCH_TERM>, <RESULTS> and <DOC_ID>; these will get replaced by the script.

Commit:
Added the package xmltodict to deal with PubMed XML responses

Commit:
Added __init__.py files to allow: import fun_3000

Commit:
#11 New script to ingest medical abstracts from pubmed

Commit:
#11 Config file for specifying different online medical db URLS

Commit:
Re-factored code to allow importing this as a module

Commit:
Re-factored code to allow importing this script as a module

Commit:
Updated config parser code to refer to correct path

Commit:
Module import changes

ayota committed:
ingestion progress so far, finished get text module, finished get_corpus.py

ayota committed:
ingestion module now done. added directory option.

lauralorenz commented:
Update to PR from @ayota at https://github.com/ayota/ddl_nlp/issues/18#issuecomment-204172936

Usage for wrapper function:

> Now to run ingestion process, run `python get_corpus.py -s [CSV OF SEARCH TERMS] -r [NUM RESULTS YOU WANT RETURNED] -d [DIRECTORY FOR ALL FETCHED FILES]`
> 
> and it will run the get abstract, get wikipedia, and get med textbook part of the process.

lauralorenz committed:
clean out dead code from medical abstracts runner, add some comments. add biopython to reqs. take out regex from pre-writing last step because that is part of a separate issue now.

lauralorenz committed:
wikipedia ingest comment

lauralorenz committed:
breaking shared method out into a utils file

lauralorenz committed:
get rid of the html strip from ingestion since cleaning is now going to be pulled in later

lauralorenz committed:
not used anywhere? I think a test script

ayota committed:
Update get_corpus.py

ayota committed:
Update med_abstract_ingest.py

deleted commented out pubmed urls since those are covered by the biopython package now.

lauralorenz committed:
Merge pull request #23 from ayota/iss11_medical_abstracts_laura

Iss11 medical abstracts laura
